Observer/CustomerSaveAfterObserver.php

```diff
@@ -30,12 +38,25 @@ public function execute(Observer $observer): void
     {
         /** @var Customer $customer */
         $customer = $observer->getEvent()->getData('customer');
+        if ($this->processedCustomersRegistry->isCustomerProcessed($customer)) {
+            return;
+        }
         $arguments = ['customerId' => $customer->getId()];
-        $data = ['customer.updated', $this->json->serialize($arguments)];
+        $data = [$this->getEventIdentifier($customer), $this->json->serialize($arguments)];
 
         $this->publisher->publish(
             QueueMetadataInterface::EVENT_QUEUE,
             $data
         );
+
+        $this->processedCustomersRegistry->setCustomerProcessed($customer);
+    }
+
+    private function getEventIdentifier(Customer $customer): string
+    {
+        if ($this->newCustomersRegistry->isCustomerNew($customer)) {
```

Comment:
You maybe able to replace this if you can determine this by comparing the `created_at` and `updated_at` values. Since the database defaults `updated_at` to `created_at` the first time the record is stored, it can guarantee that it's a new record.

Reply:
It works indeed. I changed it and removed the NewCustomersRegistry.